modules/sdk-lib-mpc/test/unit/tss/ecdsa/dklsDsg.ts

```diff
@@ -255,4 +264,149 @@ describe('DKLS Dsg 2x3', function () {
       Error('Error while creating messages from party 0, round 5: Error: k256 error: signature error')
     );
   });
+
+  it(`should get and set sesssion corretly for all rounds`, async function () {
+    const vector = vectors[0];
+
+    const party2 = new DklsDsg.Dsg(
+      fs.readFileSync(shareFiles[vector.party2]),
+      vector.party2,
+      vector.derivationPath,
+      crypto.createHash('sha256').update(Buffer.from(vector.msgToSign, 'hex')).digest()
+    );
+
+    // round 1
+    let party1Round1Message: DeserializedBroadcastMessage,
+      party2Round1Message: DeserializedBroadcastMessage,
+      party1Round1Session: string;
+    {
+      const party1 = new DklsDsg.Dsg(
+        fs.readFileSync(shareFiles[vector.party1]),
+        vector.party1,
+        vector.derivationPath,
+        crypto.createHash('sha256').update(Buffer.from(vector.msgToSign, 'hex')).digest()
+      );
+      party1Round1Message = await party1.init();
+      party1Round1Session = party1.getSession();
+
+      party2Round1Message = await party2.init();
+    }
+
+    // round 2
+    let party1Round2Messages: DeserializedMessages,
+      party2Round2Messages: DeserializedMessages,
+      party1Round2Session: string;
+    {
+      const party1Round2DSG = new DklsDsg.Dsg(
+        fs.readFileSync(shareFiles[vector.party1]),
+        vector.party1,
+        vector.derivationPath,
+        crypto.createHash('sha256').update(Buffer.from(vector.msgToSign, 'hex')).digest()
+      );
+      party1Round2DSG.setSession(party1Round1Session);
+      party1Round2Messages = party1Round2DSG.handleIncomingMessages({
+        p2pMessages: [],
+        broadcastMessages: [party2Round1Message],
+      });
+      party1Round2Session = party1Round2DSG.getSession();
+
+      party2Round2Messages = party2.handleIncomingMessages({
+        p2pMessages: [],
+        broadcastMessages: [party1Round1Message],
+      });
+    }
+
+    // round 3
+    let party1Round3Messages: DeserializedMessages,
+      party2Round3Messages: DeserializedMessages,
+      party1Round3Session: string;
+    {
+      const party1Round3DSG = new DklsDsg.Dsg(
```

Comment:
instead of having to construct a Dsg object from scratch everytime and calling setSession, i think we should support constructing a Dsg object using a function like fromSessionBytes(buffer)

Reply:
Ya i was thinking the same but since there are not exposed methods to get `keyShareBytes`, `messageHash`, `derivationPath` and `partyIdx` from the `dsgSession` (sometime they are not there yet), i was thinking in making the session payload to include all this, but that would mean that the `key` would leave external signer.